src/main/java/com/uber/cadence/internal/shadowing/ReplayWorkflowActivityImpl.java

```diff
@@ -0,0 +1,133 @@
+/*
+ *  Modifications Copyright (c) 2017-2021 Uber Technologies Inc.
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+package com.uber.cadence.internal.shadowing;
+
+import com.google.common.collect.Lists;
+import com.uber.cadence.GetWorkflowExecutionHistoryResponse;
+import com.uber.cadence.HistoryEvent;
+import com.uber.cadence.WorkflowExecution;
+import com.uber.cadence.common.WorkflowExecutionHistory;
+import com.uber.cadence.internal.common.RpcRetryer;
+import com.uber.cadence.internal.common.WorkflowExecutionUtils;
+import com.uber.cadence.internal.metrics.MetricsType;
+import com.uber.cadence.serviceclient.IWorkflowService;
+import com.uber.cadence.testing.WorkflowReplayer;
+import com.uber.m3.tally.Scope;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReplayWorkflowActivityImpl implements ReplayWorkflowActivity {
+
+  private static final Logger log = LoggerFactory.getLogger(ReplayWorkflowActivityImpl.class);
+
+  private final IWorkflowService serviceClient;
+  private final boolean failFast;
+  private final Scope metricsScope;
+  private final Class<?> workflowClass;
+  private final Class<?>[] workflowClasses;
+
+  public ReplayWorkflowActivityImpl(
+      IWorkflowService serviceClient,
+      boolean failFast,
+      Scope metricsScope,
+      Class<?> workflowClass,
+      Class<?>... workflowClasses) {
+    this.serviceClient = serviceClient;
+    this.failFast = failFast;
```

Comment:
Is it for local test? If so, shall we also include exit condition for shadowing?

Reply:
Removed this flag